mne/channels/channels.py

```diff
@@ -235,6 +237,84 @@ def get_channel_types(self, picks=None, unique=False, only_data_chs=False):
         return _get_channel_types(self.info, picks=picks, unique=unique,
                                   only_data_chs=only_data_chs)
 
+    def get_ch_positions(self, picks=None):
+        """Get a dictionary of channel positions for each channel.
+
+        Channel positions are in meters in the specified coordinate frame
+        of the montage. All channel positions are a 3-dimensional vector
+        corresponding to xyz positions in meters. For MEG channels, they
+        also add the vectorized orientation.
+
+        For more information on the different channel type
+        positions obtained, see :class:`mne.Info` documentation.
+
+        Parameters
+        ----------
+        picks : str | list | slice | None
+            None corresponds to all data channels that are not MEG.
+
+        Returns
+        -------
+        ch_positions : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding list of xyz coordinates in the montage
+             coordinate frame.
+        ch_frame_dict : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding montage coordinate frame.
+        """
+        # only get channels that were picked
+        picks = _picks_to_idx(self.info, picks)
+        chs = [self.info['chs'][ipick] for ipick in picks]
+
+        # extract name, position, orientation (for MEG)
```

Comment:
where do you get orientation?

Reply:
Oh yeah, leaving out orientation now per @larsoner  comment on https://github.com/mne-tools/mne-python/pull/7667#discussion_r481502708